### PR TITLE
Add support for database auto-discovery in the kube-agent helm chart

### PR DESCRIPTION
**Pull request by nicofff**
The kube-agent helm chart was missing the option to enable database auto-discovery. 
This PR adds a new value to values.yaml and the configmap template to pass the needed config to the agent

Note: We tested this by overriding the teleport version since at time of writing this cart still installs 7.x
Probably a good idea to release it with / after the 8.0 release

### CODE REVIEW

**Comment by nicofff** [comment]
I don't particularly like the `db_autodiscovery` name, but a top level `aws` value seemed worse. Open to change it to whatever you feel is best

File: examples/chart/teleport-kube-agent/values.yaml [hunk 1]
```diff
@@ -39,6 +39,18 @@ apps: []
...
+# Optional values if Database access is enabled.
+################################################################
+
+# Autodiscovery settings. Example:
+# aws_autodiscovery:
+# - types: ["rds"]
+#   regions: ["us-west-1"] # AWS regions to fetch databases from
+#   tags: # AWS database resource tags to match
+#     "*": "*"
+db_autodiscovery: []
```
Comment (r0mant) [inline examples/chart/teleport-kube-agent/values.yaml:52]: How about something like

```yaml
discovery:
  databases:
    aws:
    - types: ["rds"]
      regions: ["us-west-1"]
      tags:
        "*": "*"
```

This is a bit more nested structure but will allow us to have a place to extend this in the future to other things that can be auto-discovered like apps (e.g. on K8s), desktops or databases in other clouds.

WDYT @nicofff @webvictim?

**Review by r0mant — COMMENTED** [review]
@nicofff Thank you for the contribution! I have left a suggestion regarding the discovery values structure.

File: examples/chart/teleport-kube-agent/values.yaml (hunk 1 above)
Comment (webvictim) [reply examples/chart/teleport-kube-agent/values.yaml:52]: Yes, I agree this is a good way to go about it.

**Review by webvictim — CHANGES_REQUESTED** [review]
@nicofff Thanks for the PR. Looks good overall, but there are a few requests before we can merge (as well as @r0mant's comment):

1. Please add `discovery` at the appropriate location in the `teleport-kube-agent` chart reference: https://github.com/gravitational/teleport/blob/master/docs/pages/kubernetes-access/helm/reference.mdx

2. Please add `discovery` to https://github.com/gravitational/teleport/blob/master/examples/chart/teleport-kube-agent/values.schema.json

3. Please add an example which sets `discovery` so it can be validated by the linter - see other examples in https://github.com/gravitational/teleport/tree/master/examples/chart/teleport-kube-agent/.lint

Thank you!

**Comment by russjones** [comment]
@nicofff Ping on this issue, are you planning on making these changes? If not we can assign someone internally to take over this PR.

**Comment by nicofff** [comment]
Hey @russjones, haven't had time to look into this, and probably won't in the near future. So more than happy for you guys to take over

**Comment by webvictim** [comment]
Think this is going to be superseded by https://github.com/gravitational/teleport/pull/9857 so will close it.

**Pull request closed without merging**